media/js/invokeRemoteRestApiVue.js

```diff
@@ -0,0 +1,68 @@
+console.log('Loaded!')
+;(function() {
+    const vscode = acquireVsCodeApi()
+    const app = new Vue({
+        el: '#app',
+        data: {
+            selectedApiResource: '',
+            selectedMethod: '',
+            methods: [],
+            jsonInput: '',
+            queryString: '',
+            errors: [],
+            isLoading: false,
+        },
+        mounted() {
+            this.$nextTick(function() {
+                window.addEventListener('message', this.handleMessageReceived)
+            })
+        },
+        methods: {
+            setApiResource: function() {
+                vscode.postMessage({
+                    command: 'apiResourceSelected',
+                    value: this.selectedApiResource,
+                })
+            },
+            handleMessageReceived: function(e) {
+                const message = event.data
+                console.log(message.command)
+                switch (message.command) {
+                    case 'setMethods':
+                        this.methods = message.methods
+                        if (this.methods) {
+                            this.selectedMethod = this.methods[0]
+                        }
+                        break
+                    case 'invokeApiStarted':
+                        this.isLoading = true
+                        break
+                    case 'invokeApiFinished':
+                        this.isLoading = false
+                        break
+                }
+            },
+            sendInput: function() {
+                this.errors = []
+                if (!this.selectedApiResource && !this.selectedMethod) {
+                    if (!this.selectedApiResource) {
+                        this.errors.push('Please select an API resource')
```

Comment:
This needs to be localized?

Reply:
also "Please" is not needed in messages, it distracts from the main topic of the message

Reply:
updated with the message passing needed to make localization work